gmime/gmime-utils.c

```diff
@@ -1468,7 +1469,9 @@ rfc2047_token_new_encoded_word (const char *word, size_t len)
 	token = rfc2047_token_new (payload, inptr - payload);
 	token->charset = g_mime_charset_iconv_name (charset);
 	token->encoding = encoding;
-	
+	/* RFC 2047 forbids SP in the encoded-word */
+	token->sp_in_encword = memchr(token->text, ' ', token->length) ? 1 : 0;
+
```

Comment:
It forbids more than just spaces... there's a list of special characters that are not allowed (and it also depends on whether you are encoding a `phrase` vs unstructured text).

The `tokenize_rfc2047_phrase` and `tokenize_rfc2047_text` functions have STRICT vs LOOSE branches. The STRICT branch is already strict enough that it won't allow spaces in an encoded-word token.

The LOOSE branch allows pretty much any slop that looks vaguely like an encoded-word token.

A simpler/less expensive way of doing this would just be to have those methods keep an extra bool state variable to say whether it is "valid" or not and then only warn if `if ((token = rfc2047_token_new_encoded_word (word, n))) {` evaluates to true.

I would probably also just warn in the tokenize functions rather than tokenizing and then scanning the tokens to see if any of them have sp_in_encword set.

In other words...

```c
if ((token = rfc2047_token_new_encoded_word (word, n))) {
    if (has_specials)
        _g_mime_parser_options_warn (options, offset, GMIME_WARN_INVALID_RFC2047_HEADER_VALUE, text);

    ...
```

Reply:
Yes, that's a much better approach.  Will fix that, too…

BTW, I think it's about time to send you a big __thank you__ for your great work and this extremely useful library!  And of course I wish you a happy Christmas and all the best for the new year!

Reply:
Don't worry about it, I've already merged your patch and made the adjustments myself.

And thanks! :)

Merry Christmas and a happy New Year to you as well!